### PR TITLE
add properties.operation definition (#47)

### PR DESCRIPTION
**Pull request by tomkralidis**
Addresses #47

### CODE REVIEW

File: schemas/notificationMessageGeoJSON.yaml [hunk 1]
```diff
@@ -94,6 +94,14 @@ properties:
...
     type: object
     properties:
+      operation:
+        type: string
+        description: the event associated with the lifecycle of a resource.
+        enum:
+          - create
+          - update
+          - delete
+        default: created
```
Comment (golfvert) [inline schemas/notificationMessageGeoJSON.yaml:104]: Should be "create" not "created"

File: standard/requirements/core/REQ_operation.adoc [hunk 2]
```diff
@@ -0,0 +1,7 @@
+[[req_core_operation]]
+[width="90%",cols="2,6a"]
+|===
+^|*Recommendation {counter:req-id}* |*/req/core/operation*
+^|A |A WNM SHALL provide the `+properties.operation+` property to indicate if a resource has been updated or deleted.
+^|B |For an update or delete operation, a WNM SHALL provide one link object with `+rel=canonical+`, and `+href=#+`.
```
Comment (golfvert) [inline standard/requirements/core/REQ_operation.adoc:6]: For "delete" yes. The link object is (sort of) empty like that.
For "update", it SHALL provide a new download link.

**Review by golfvert — APPROVED** [review]

File: schemas/notificationMessageGeoJSON.yaml (hunk 1 above)
Comment (tomkralidis) [reply schemas/notificationMessageGeoJSON.yaml]: fixed

File: standard/requirements/core/REQ_operation.adoc (hunk 2 above)
Comment (tomkralidis) [reply standard/requirements/core/REQ_operation.adoc]: Updated so that empty links are ONLY for delete.

**Comment by josusky** [comment]
Is this `operation` property expected to be used only for metadata records or also for data instances? Because if it is only for metadata we should clearly state so and perhaps also remove the `data_id` from such examples. If it is really meant also for data instances (files) then there should be more discussion about its meaning in such a use case.

**Comment by tomkralidis** [comment]
Closing in lieu of https://github.com/wmo-im/wis2-notification-message/issues/47#issuecomment-1695689179

**Pull request closed without merging**